### DISCUSSION TITLE
[DataX]插件 从mysql读取数据写入sqlserver-2012 报 字段错误

### DISCUSSION THREAD

**Discussion by sakinsun** [apache/dolphinscheduler#6453] [Q&A]
我在DS 1.3.6版本中，使用datax插件，数据源为mysql，目标源为sqlserver，使用的是 "非自定义模板"的方式，就是 mysql的地方写sql语句，目标sqlserver地方直接填写表名，然后运行这个节点。结果运行报错，报错内容如下：
”Description:[获取表字段相关信息失败.].  - 获取表:test_tmp_20211001 的字段的元信息时失败. 请联系 DBA 核查该库、表信息. - com.microsoft.sqlserver.jdbc.SQLServerException: “`”附近有语法错误。”

DataX在DS下运行后，日志中可以看到在获取sqlserver元数据字段的时候是这样的：
"writer":{
					"name":"sqlserverwriter",
					"parameter":{
						"column":[
							"`suppid`"   <--- (这里双引号里面有2个间隔号)
						]

sqlserver表字段是用 ` 这个号引起来的，导致在sqlserver里面查询元数据报错，DS源码如下：
switch (dbType) {
            case MYSQL:
                return String.format("`%s`", column);
            case POSTGRESQL:
                return String.format("\"%s\"", column);
            case ORACLE:
                return String.format("\"%s\"", column);
            case SQLSERVER:
                return String.format("`%s`", column);   <--------- 就是这里！！！！
            default:
                return column;
        }
    }

源码所在github位置：dolphinscheduler-task-plugin/dolphinscheduler-task-datax/src/main/java/org/apache/dolphinscheduler/plugin/task/datax/DataxUtils.java

**Comment by sakinsun** [comment]
https://github.com/sakinsun/pic/blob/main/sqlserverwriter-column.png

github上的 ` 间隔号无法正常显示，所以我把ds上的日志截图了

**Reply by KillerXV** [reply]
你好，遇到同样的问题，请问解决了吗
